health-services/resource-estimation-service/src/main/java/org/egov/processor/service/ResourceEstimationService.java

```diff
@@ -0,0 +1,87 @@
+
+import org.egov.processor.config.Configuration;
+import org.egov.processor.config.ServiceConstants;
+import org.egov.processor.repository.ServiceRequestRepository;
+import org.egov.processor.util.CampaignIntegrationUtil;
+import org.egov.processor.web.models.File;
+import org.egov.processor.web.models.PlanConfiguration;
+import org.egov.processor.web.models.PlanConfigurationRequest;
+import org.egov.processor.web.models.campaignManager.CampaignSearchRequest;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class ResourceEstimationService {
+
+    private final FileParser excelParser;
+    private final FileParser geoJsonParser;
+    private final FileParser shapeFileParser;
+    private CampaignIntegrationUtil campaignIntegrationUtil;
+	private ServiceRequestRepository serviceRequestRepository;
+	private Configuration config;
+
+	public ResourceEstimationService(FileParser excelParser, FileParser geoJsonParser, FileParser shapeFileParser,CampaignIntegrationUtil campaignIntegrationUtil
+    		,ServiceRequestRepository serviceRequestRepository,
+    		Configuration config) {
+        this.excelParser = excelParser;
+        this.geoJsonParser = geoJsonParser;
+        this.shapeFileParser = shapeFileParser;
+        this.campaignIntegrationUtil= campaignIntegrationUtil;
+    	this.serviceRequestRepository=serviceRequestRepository;
+    	this.config=config;
+    }
+
+	/**
+	 * Estimates resources required for the plan configuration by parsing files and fetching campaign search results.
+	 *
+	 * @param planConfigurationRequest The plan configuration request containing necessary information for estimating resources.
+	 */
+    public void estimateResources(PlanConfigurationRequest planConfigurationRequest) {
+        PlanConfiguration planConfiguration = planConfigurationRequest.getPlanConfiguration();
+
+        Map<File.InputFileTypeEnum, FileParser> parserMap = getInputFileTypeMap();
+        CampaignSearchRequest campaignRequest = campaignIntegrationUtil.buildCampaignRequestForSearch(planConfigurationRequest);
+        Object campaignSearchResponse = serviceRequestRepository.fetchResult(new StringBuilder(config.getProjectFactoryHostEndPoint()+config.getCampaignIntegrationSearchEndPoint()),
+				campaignRequest);
+        processFiles(planConfigurationRequest, planConfiguration, parserMap, campaignSearchResponse);
+    }
+
+	private void processFiles(PlanConfigurationRequest planConfigurationRequest, PlanConfiguration planConfiguration,
+			Map<File.InputFileTypeEnum, FileParser> parserMap, Object campaignSearchResponse) {
+		for(File file:planConfiguration.getFiles())
+        {
+            if(file.getActive())
+            {
+                File.InputFileTypeEnum fileType = file.getInputFileType();
+                FileParser parser = parserMap.computeIfAbsent(fileType, ft -> {
+                    throw new IllegalArgumentException("Unsupported file type: " + ft);
```

Comment:
Is this how this is supposed to work?  if one of the file is invalid then throw error for the whole flow ?

Reply:
Yes as we are handling specific types only and if its not present into that then we have to throw exception.